EIPS/eip-6384.md

```diff
@@ -0,0 +1,155 @@
+
+### Function's inputs
+
+The inputs of the function:
+
+- `domainHash` is the EIP-712's domainSeparator, a hashed `eip712Domain` struct.
+- `primaryType`is the EIP-712's `primaryType`.
+- `encodedSignature` is an ABI encoded message part of the EIP-712 full message.
+  We want to get the feedback of the community and specifically smart contract implementors on how they would like to receive this information: e.g. as seperate parameters as we show here or the fully ABI encoded message that includes all EIP-712 elements (including `types`).
+
+### Function's output(s)
+
+The output of the the function is an array of strings. The wallet SHOULD display them to its end-users. The wallet MAY choose to augment the returned strings with additional data. (e.g. resolve contract addresses to their name)
+
+The strings SHOULD NOT be formatted (e.g. should not contain HTML code) and wallets SHOULD treat this string as an untrusted input and handle its rendering as such.
+
+As it might be useful to provide a more structured output, we want to get the feedback of the community and specifically wallets' implementors on how they would like to receive it. One suggestion is that function would return structs that represent transfers and approvals to support a simulation like experience to the users. e.g. the `evalEIP712Buffer` function would return a strcut that represnts a transfer (contract, from, to, amount) and the wallet MAY display a resolved version of it to the end user. ("transfer 2 USDC to address 0x...")
+
+### Support for EIP-712 messages that are not meant to be used on-chain
+
+In case the EIP-712 message is not bounded to a contract, the wallet cannot call this function as the contract address is unkown and SHOULD default to its current display.
+
+Some web2 implementations are using a signed EIP-712 message as a method to identify the user. It would be helpful if the wallet can display a reassuring message that this message cannot be used on-chain.
+We would like to get feedback from the community if this can be done safely, e.g. by omitting the `chainId` and/or `verifyingContract` from EIP-712 message, or putting non-exisiting values in them (e.g. `verifyingContract` = 0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC)
+
+## Rationale
+
+The proposed solution solves the readability issues. The incentives for keeping the EIP-712 message descritption as accurate as possible are alligned, as the responsibility for the description is now owned by the contract, that:
+
+- knows the message meaning exactly (and probably can reuse the code that handles this message when received on chain)
+- Natively incentivized to provide the best explanation to prevent a possible fraud
+- Not involving a third party that needs to be trusted
+- Maintains the fee-less customer experience as the added function is in “view” mode and does not require an on-chain execution and fees.
+- Maintains Web3’s composability property
```

Comment:
This is more like Motivation than Rationale. The Rationale section should describe individual technical choices made in your EIP, while the Motivation should justify the existence of the EIP as a whole.

A great example of something that should belong in your Rationale section is why your function returns an array of strings.

Reply:
updated accordingly :)